### PR TITLE
add kubernetes 1.13

### PR DESCRIPTION
**Pull request by JulienBalestra**
### What does this PR do?

Introduce Kubernetes 1.13 in pupernetes.

### Additional Notes

The scheduler is now responsible to start daemonsets, so it's not possible anymore to use the controller manager to start a scheduler as daemonset.

### CODE REVIEW

File: pkg/setup/templates/1.13.go [hunk 1]
```diff
@@ -0,0 +1,719 @@
...
+    - --service-account-private-key-file=/etc/secrets/service-accounts.rsa
+    - --concurrent-deployment-syncs=2
+    - --concurrent-endpoint-syncs=2
+    - --concurrent-gc-syncs=5
+    - --concurrent-namespace-syncs=3
+    - --concurrent-replicaset-syncs=2
+    - --concurrent-resource-quota-syncs=2
+    - --concurrent-service-syncs=1
+    - --concurrent-serviceaccount-token-syncs=2
+    - --horizontal-pod-autoscaler-use-rest-clients=true
```
Comment (CharlyF) [inline pkg/setup/templates/1.13.go]: This option is deprecated in 1.13 https://github.com/kubernetes/kubernetes/commit/8ef369ea2d1f487e46ca90f64dd30111eafafa78

File: pkg/setup/templates/1.13.go [hunk 2]
```diff
@@ -0,0 +1,719 @@
...
+		{
+			Name:        "kube-controller-manager.yaml",
+			Destination: ManifestAPI,
+			Content: []byte(`---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kube-controller-manager
+  namespace: kube-system
+automountServiceAccountToken: false
```
Comment (CharlyF) [inline pkg/setup/templates/1.13.go]: In the logs of the controller manager I see:

```
W0104 14:13:57.716813       1 authentication.go:373] failed to read in-cluster kubeconfig for delegated authentication: failed to read token file "/var/run/secrets/kubernetes.io/serviceaccount/token": open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
W0104 14:13:57.716904       1 authentication.go:235] No authentication-kubeconfig provided in order to lookup client-ca-file in configmap/extension-apiserver-authentication in kube-system, so client certificate authentication won't work.
W0104 14:13:57.716925       1 authentication.go:238] No authentication-kubeconfig provided in order to lookup requestheader-client-ca-file in configmap/extension-apiserver-authentication in kube-system, so request-header client certificate authentication won't work.
W0104 14:13:57.716952       1 authorization.go:177] failed to read in-cluster kubeconfig for delegated authorization: failed to read token file "/var/run/secrets/kubernetes.io/serviceaccount/token": open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
W0104 14:13:57.716987       1 authorization.go:146] No authorization-kubeconfig provided, so SubjectAccessReview of authorization tokens won't work.
[...]
```

Not sure if this is a result of this option ?

File: pkg/setup/templates/1.13.go [hunk 3]
```diff
@@ -0,0 +1,719 @@
...
+  template:
+    metadata:
+      labels:
+        dns: coredns
+    spec:
+      serviceAccountName: coredns
+      tolerations:
+        - key: "CriticalAddonsOnly"
+          operator: "Exists"
+      containers:
```
Comment (CharlyF) [inline pkg/setup/templates/1.13.go]: I had restarts on my coredns pod (might be due to my setup), the logs are very limited. I could not find ways to have more verbose logs, do you know of any option ?

File: pkg/setup/templates/1.13.go (hunk 1 above)
Comment (JulienBalestra) [reply pkg/setup/templates/1.13.go]: Thanks for the highlight.

File: pkg/setup/templates/1.13.go (hunk 3 above)
Comment (JulienBalestra) [reply pkg/setup/templates/1.13.go]: You can alter the behaviour of coredns with the debug plugin, [see](https://github.com/coredns/coredns/tree/master/plugin/debug).
Let me know if it's related to your setup or with pupernetes itself.

File: pkg/setup/templates/1.13.go (hunk 2 above)
Comment (JulienBalestra) [reply pkg/setup/templates/1.13.go]: Oh this is interesting, I never noticed it before.
I'm going to mount the certificates and a proper kubeconfig to observe to settle this.
The subjectAccessReview is an important feature, but I wonder how the controller manager uses it.

File: pkg/setup/templates/1.13.go (hunk 3 above)
Comment (CharlyF) [reply pkg/setup/templates/1.13.go]: Haven't been able to get it to restart again. I will keep this option in mind though.

**Pull request merged**